lib/plugins/aws/deployFunction.js

```diff
@@ -76,6 +78,9 @@ class AwsDeployFunction {
           await this.deployFunction();
         }
         await this.updateFunctionConfiguration();
+        if (this._shouldEnsureFunctionState) {
+          await this.ensureFunctionState();
```

Comment:
I thought the issue on our side is that we need to confirm on function state after `this.deployFunction()` is done and before doing `this.updateFunctionConfiguration()` (as it may fail if it's not settled).

While all we do in this PR is simply prolonging command execution so it doesn't exit until the function state settles. If that's the case, then the only thing we're fixing is that following up with another command won't crash due to function being in interim state due to effect of previous command. Is that correct?

Reply:
I thought the same thing, but after finally being able to reproduce this issue, I've realized and noticed that in recent versions of the Framework, the retry for `updateFunctionConfiguration` is already handling that very well so there's no need to add any extra steps which will pretty much do the same thing.

> While all we do in this PR is simply prolonging command execution so it doesn't exit until the function state settles. If that's the case, then the only thing we're fixing is that following up with another command won't crash due to function being in interim state due to effect of previous command. Is that correct?

Correct 👍